File: salimmo-parent/property-service/src/main/java/org/propertyservice/entities/Property.java
package org.propertyservice.entities;


import lombok.*;
import org.hibernate.annotations.Where;
import org.propertyservice.enums.PropertyCategory;
import org.propertyservice.enums.PropertyStatus;
import org.propertyservice.enums.PropertyType;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Table(name = "property")
@Where(clause = "deleted=false")
public class Property extends BaseEntity implements Serializable {
    private UUID ref;
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "owner_id")
    private PropertyOwner owner;
    @Enumerated(EnumType.STRING)
    private PropertyCategory category;
    @Enumerated(EnumType.STRING)
    private PropertyType type;
    @OneToOne
    private PropertyLocation propertyLocation;
    @OneToOne
    private InnerProperty innerProperty;
    @OneToOne
    private PropertySurface propertySurface;
    @OneToOne
    private PropertyEnergies propertyEnergies;
    @OneToOne
    private ExteriorProperty exteriorProperty;
    private String title;
    private String description;
    private Double fees;
    private Double netPrice;
    private Byte [] images;
    private PropertyStatus status;
    private String createdBy;
    private String updatedBy;
    private String deletedBy;
    private Boolean deleted=false;
    private LocalDateTime deletedAt;

    @PostPersist
    private void generateRefer(){
        this.ref = UUID.randomUUID();
    }
}

File: salimmo-frontend/src/app/models/property/property.ts
import {PropertyOwner} from "../propertyowner/propertyOwner";
import {PropertyCategory} from "../enums/property-category";
import {PropertyType} from "../enums/property-type";
import {Propertylocation} from "../propertylocation/propertylocation";
import {Innerproperty} from "../innerproperty/innerproperty";
import {Propertysurface} from "../propertysurface/propertysurface";
import {Propertyenrgies} from "../propertyenergies/propertyenrgies";
import {Exteriorproperty} from "../exteriorproperty/exteriorproperty";

export class Property {
  id!:number;
  ref!:string;
  owner!:PropertyOwner;
  category!:PropertyCategory;
  type!:PropertyType;
  propertyLocation!:Propertylocation;
  innerProperty!:Innerproperty;
  propertySurface!:Propertysurface;
  propertyEnergies!:Propertyenrgies;
  exteriorProperty!:Exteriorproperty;
  title!:string;
  description!:string;
  fees!:number;
  netPrice!:number;
  images!:string[];
}
